tests/test_synced_collections/test_mongodb_collection.py

```diff
@@ -29,49 +28,73 @@
     PYMONGO = False
 
 
+try:
+    import numpy
```

Comment:
Let's be consistent in using `numpy` or `np`, but not both. It makes a huge difference when refactoring. At one point, I refactored all uses of `np` to `numpy`, to make the codebase consistent with its pre-existing choice of `numpy`. I noticed a few `np` uses earlier in this PR.